src/components/descriptions/DescriptionsKeyHandler.ts

```diff
@@ -16,38 +16,34 @@ type AutoDescriptionsProps = {
  * Handles the longer and shorter description change when Alt (option) + B or Alt (option) + S are pressed, respectively.
  *
  * @export
- * @param {React.RefObject<HTMLElement>} ref
- * @param {string} type
- * @param {string[]} descs
- * @param {string} title
- * @param {AutoDescriptionsProps} [autoDescriptions]
- * @param {React.KeyboardEvent} [event]
- * @return {void}
  */
-export function descriptionsChanger(
-	ref: React.RefObject<HTMLElement>,
-	type: string,
-	descs: string[],
-	title: string,
-	autoDescriptions?: AutoDescriptionsProps,
-	event?: React.KeyboardEvent,
-): void {
+export function descriptionsKeyHandler({
+	chartRef,
+	setDescriptionContent,
+	type,
+	descs,
+	title,
+	autoDescriptions,
+	event,
+}: {
```

Comment:
Move to an interface.

Reply:
```suggestion
/**
 * This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, version 3 of the License.
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
 * You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
 * Other licensing options may be available, please reach out to data-viz@feedzai.com for more information.
 */

type AutoDescriptionsProps = {
  dynamicDescriptions?: boolean;
  apiKey: string;
  model?: string;
  temperature?: number;
};

interface DescriptionsKeyHandlerProps {
  chartRef: React.RefObject<HTMLElement>;
  setDescriptionContent: (content: string) => void;
  type: string;
  descs: [string, string];
  title: string;
  autoDescriptions?: AutoDescriptionsProps;
  event?: React.KeyboardEvent;
}

/**
 * Handles the longer and shorter description change when Alt (option) + B or Alt (option) + S are pressed, respectively.
 */
export function descriptionsKeyHandler({
  chartRef,
  setDescriptionContent,
  type,
  descs,
  title,
  autoDescriptions,
  event,
}: DescriptionsKeyHandlerProps): void {
  if (!chartRef.current) return;

  const fullTitle = `${title}, ${type}. ${autoDescriptions ? "Automatic description: " : ""}`;

  if (!event) {
    setDescriptionContent(fullTitle + descs[1]);
    return;
  }

  if (event.altKey) {
    switch (event.code) {
      case "KeyS":
        setDescriptionContent(fullTitle + descs[1]);
        break;
      case "KeyB":
        setDescriptionContent(fullTitle + descs[0]);
        break;
    }
  }
}
```